src/pybind/mgr/dashboard/rest_client.py

```diff
@@ -26,6 +27,16 @@
     from urllib3.exceptions import SSLError
 
 
+class TimeoutRequestsSession(requests.Session):
+    """
+    Set timeout argument for all requests if this is not already done.
+    """
+    def request(self, *args, **kwargs):
+        if kwargs.get('timeout') is None and Settings.REST_REQUESTS_TIMEOUT > 0:
```

Comment:
This is IMO too generic. Is there any usage, where we explicitly change the timeout for a specific request? If not, why not simply call
```python
super(TimeoutRequestsSession, self).request(*args, **kwargs, timeout=Settings.REST_REQUESTS_TIMEOUT)
```
?